lighthouse-core/lib/tracehouse/trace-processor.js

```diff
@@ -78,11 +136,30 @@ class TraceProcessor {
       }
     }
 
-    // sort by ts, if there's no ts difference sort by index
-    indices.sort((indexA, indexB) => {
-      const result = traceEvents[indexA].ts - traceEvents[indexB].ts;
-      return result ? result : indexA - indexB;
-    });
+    // Sort by ascending timestamp first.
+    indices.sort((indexA, indexB) => traceEvents[indexA].ts - traceEvents[indexB].ts);
+
+    // Now we find groups with equal timestamps and order them by their nesting structure.
+    for (let i = 0; i < indices.length - 1; i++) {
```

Comment:
this can increment `i` up by `tsGroupIndices.length`, not just 1 (otherwise lots of repeat sorts of already sorted indices)